Sources/GRPCProtobufCodeGen/ProtobufCodeGenParser.swift

```diff
@@ -166,3 +171,22 @@ extension FileDescriptor {
     return header
   }
 }
+
+extension SwiftProtobufNamer {
+  internal func trimTrailingUnderscores(from name: String) -> String {
+    var nameCopy = name
+    while nameCopy.hasSuffix("_") {
+      nameCopy = String(nameCopy.dropLast())
+    }
+    return nameCopy
```

Comment:
We copy the string for each `_` we drop. Instead we can look for the last index in the string which isn't an `_` (if there's no such index then the string only contains `_` or is empty in which case we can just return an empty string):

```swift
if let index = name.lastIndex({ $0 != "_" }) {
  return String(name[...index])
} else {
  return ""
}
```

This would also make more sense as an extension on `String` (which could take the `Character` to trim).